src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,84 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  selectedTodo: Todo,
+  setSelectedTodo: (a:Todo | null) => void,
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const handleCloseButton = () => setSelectedTodo(null);
+  const {
+    title,
+    userId,
+    id,
+    completed,
+  } = selectedTodo;
+
+  useEffect(() => {
+    const getUserFromServer = async () => {
+      const UserFromServer = await getUser(userId);
+
+      setUser(UserFromServer);
+    };
+
+    getUserFromServer();
+  }, [selectedTodo]);
```

Comment:
Good, but I'm pretty sure that you don't need `selectedTodo` in dependency array. Modal in this case only mount and unmount and never update, so you don't need any dependencies in your array.
```suggestion
  useEffect(() => {
    const getUserFromServer = async () => {
      const UserFromServer = await getUser(userId);

      setUser(UserFromServer);
    };

    getUserFromServer();
  }, []);
```